src/normalizer/Normalizer.zig

```diff
@@ -21,31 +24,18 @@ decomp_trie: Trieton,
 
 const Self = @This();
 
-/// Form is the normalization form.
-/// * .canon : Canonical decomposition, which always results in two code points.
-/// * .compat : Compatibility decomposition, which can result in at most 18 code points.
-/// * .same : Default canonical decomposition to the code point itself.
-pub const Form = enum {
-    canon, // D
-    compat, // KD
-    same, // no more decomposition.
-};
-
-/// Decomp is the result of decomposing a code point to a normaliztion form.
-pub const Decomp = struct {
-    form: Form = .canon,
-    len: usize = 2,
-    seq: [18]u21 = [_]u21{0} ** 18,
-};
-
 pub fn init(allocator: *mem.Allocator, filename: []const u8) !Self {
     var self = Self{
         .allocator = allocator,
         .arena = std.heap.ArenaAllocator.init(allocator),
         .decomp_trie = Trieton.init(allocator),
     };
 
-    try self.load(filename);
+    var file = try DecompFile.decompressFile(allocator, filename);
+    defer file.deinit();
+    while (file.next()) |entry| {
+        try self.decomp_trie.add(entry.key[0..entry.key_len], entry.value);
```

Comment:
Note that since the file parsing is now internally storing the entries as a `std.ArrayList`, memory consumption before it is released will be a little higher than before this change. My hope is to remove that later on and make `DecompFile` a proper iterative implementation, backed either by the raw `UnicodeData.txt` file or by the compressed variant. For now, it's just a stepping stone and I imagine not too much additional memory.